packages/mobile/src/qrcode/QRScanner.tsx

```diff
@@ -1,176 +1,95 @@
-import { headerWithBackButton } from 'src/navigator/Headers'
-import { navigate } from 'src/navigator/NavigationService'
+import { useSafeArea } from 'react-native-safe-area-view'
+import { Defs, Mask, Rect, Svg } from 'react-native-svg'
+import { useDispatch } from 'react-redux'
+import { Namespaces } from 'src/i18n'
 import { Screens } from 'src/navigator/Screens'
-import { StackParamList } from 'src/navigator/types'
+import { QRTabParamList } from 'src/navigator/types'
 import NotAuthorizedView from 'src/qrcode/NotAuthorizedView'
-import { RootState } from 'src/redux/reducers'
 import { handleBarcodeDetected, QrCode } from 'src/send/actions'
-import { TransactionDataInput } from 'src/send/SendAmount'
 import Logger from 'src/utils/Logger'
 
-interface StateProps {
-  scanIsForSecureSend?: true
-  transactionData?: TransactionDataInput
-}
-
-interface DispatchProps {
-  handleBarcodeDetected: typeof handleBarcodeDetected
-}
+type Props = StackScreenProps<QRTabParamList, Screens.QRScanner>
 
-type OwnProps = StackScreenProps<StackParamList, Screens.QRScanner> & { isFocused: boolean }
-type Props = DispatchProps & WithTranslation & StateProps & OwnProps
+const SeeThroughOverlay = () => {
+  const { width, height } = variables
 
-const mapStateToProps = (state: RootState, ownProps: OwnProps): StateProps => {
-  const { route } = ownProps
-  return {
-    scanIsForSecureSend: route.params?.scanIsForSecureSend,
-    transactionData: route.params?.transactionData,
-  }
-}
+  const margin = 40
+  const centerBoxSize = width - margin * 2
+  const centerBoxBorderRadius = 8
 
-const mapDispatchToProps = {
-  handleBarcodeDetected,
-}
-
-const QRScannerWrapper = (props: Props) => {
-  const isFocused = useIsFocused()
-  return <QRScanner {...props} isFocused={isFocused} />
+  // TODO(jeanregisser): Investigate why the mask is pixelated on iOS.
+  // It's visible on the rounded corners but since they are small, I'm ignoring it for now.
+  return (
+    <Svg height={height} width={width} viewBox={`0 0 ${width} ${height}`}>
+      <Defs>
+        <Mask id="mask" x="0" y="0" height="100%" width="100%">
+          <Rect height="100%" width="100%" fill="#fff" />
+          <Rect
+            x={margin}
+            y={(height - centerBoxSize) / 2}
+            rx={centerBoxBorderRadius}
+            ry={centerBoxBorderRadius}
+            width={centerBoxSize}
+            height={centerBoxSize}
+            fill="#000"
+          />
+        </Mask>
+      </Defs>
+      <Rect height="100%" width="100%" fill="rgba(0,0,0,0.5)" mask="url(#mask)" />
+    </Svg>
+  )
 }
 
-class QRScanner extends React.Component<Props> {
-  static navigationOptions = () => ({
-    ...headerWithBackButton,
-    headerTitle: i18n.t('sendFlow7:scanCode'),
-  })
+export default function QRScanner({ route }: Props) {
+  const dispatch = useDispatch()
+  const { t } = useTranslation(Namespaces.sendFlow7)
+  const inset = useSafeArea()
```

Comment:
I'm wondering if this would work better as an HOC instead of getting the inset and manually using it in margins like we do here.

Reply:
I'm not sure I get the concern here.
I initially used `SafeAreaView`, but it was causing a visual glitch in the animation because of its use of `onLayout`.